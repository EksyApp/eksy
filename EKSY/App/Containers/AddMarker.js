import React, {Component} from 'react'
import {View, Text, TextInput, StyleSheet} from 'react-native'
// import {Container, Content, Header, Body, Title, Left, Button} from 'native-base'
import PointSelector from '../Components/PointSelector'
import Marker from '../Map/Marker'
import MapManager from '../Map/MapManager'
import Header from '../Components/Header'
import * as ReduxActions from '../Actions'
import {Actions} from 'react-native-router-flux'
import {connect} from 'react-redux'
import {backgroundColor, frontgroundColor} from '../Theme'
import {Grid, Row, FormInput, Button} from 'react-native-elements'
import ActionButton from '../Components/ActionButton'

class AddMarker extends Component {
	
	constructor(props) {
		super(props)
		
		this.state = {
			text: ""
		}
		
		this.mapManager = new MapManager();
	}
	
	addNewMarker() {
		let latitude = this.props.currentRegion.latitude;
		let longitude = this.props.currentRegion.longitude;
		let marker = <Marker latitude={latitude} longitude={longitude} text={this.state.text}/>;
		this.mapManager.addMarker(marker);
		setTimeout(() => this.mapManager.flyToPosition(latitude, longitude), 1000)
		Actions.mapView();
		// this.mapManager.update();
	}
	
	
	render() {
		return (
				// <Container>
				//   <Header>
				//
				//     <MenuButton />
				//     <Left />
				//     <Body>
				//       <Title>Add a new marker</Title>
				//     </Body>
				//   </Header>
				//   <Content style={Style.container}>
				//     <PointSelector onChange={(region) => this.state.region = region} style={Style.map} />
				//     <TextInput
				//       onChange={(event) => this.state.text = event.nativeEvent.text}
				//       multiline
				//       numberOfLines={10}
				//       style = {Style.textInput}
				//     />
				//     <Button block onPress={() => this.addNewMarker()} style={Style.button}>
				//       <Text style={Style.text}>Add</Text>
				//     </Button>
				//   </Content>
				//
				// </Container>
				
				<View style={styles.container}>
					<Header title="Add Marker" menuButtonPress={this.props.menuButtonPress}/>
					<Grid>
						<Row size={2} containerStyle={styles.row}>
							<PointSelector onChange={(region) => {
								this.props.regionChange(region)
							}} currentRegion={this.props.currentRegion} style={styles.map}/>
						</Row>
						<Row size={2} containerStyle={styles.row}>
							<FormInput onChangeText={(text) => this.setState({text: text})} multiline numberOfLines={8} style={styles.textArea}/>
						</Row>
						<Row size={1} containerStyle={styles.row}>
							<ActionButton title="Add" onPress={() => {
								this.addNewMarker()
							}}/>
						</Row>
					</Grid>
				</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		backgroundColor: backgroundColor,
		flex: 1
	},
	
	row: {
		width: '100%'
	},
	
	map: {
		flex: 1,
	},
	
	textArea: {
		flex: 1,
		textAlignVertical: 'top',
		width: '100%',
		backgroundColor: frontgroundColor,
		marginTop: 10
	}
	
	
})

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		menuButtonPress: () => {
			dispatch(ReduxActions.drawerOpen())
		},
		regionChange: (region) => {
			dispatch(ReduxActions.updateRegion(region))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(AddMarker)
