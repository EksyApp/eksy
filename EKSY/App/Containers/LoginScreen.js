
import {
  Text,
  View,
  StyleSheet,
  dismissKeyboard,
  TouchableWithoutFeedback
} from 'react-native'

import {
    Grid,
    Row
} from 'react-native-elements'

import Header from '../Components/Header'
import MenuButton from '../Components/MenuButton'
import Button from '../Components/Button'
import Input from '../Components/Input'
import React, {Component} from 'react'
import * as firebase from 'firebase'
import DismissKeyboard from 'dismissKeyboard'
import PostOffice from '../lib/PostOffice'
import {connect} from 'react-redux'
import * as Actions from '../Actions'
import * as Theme from '../Theme'

class LoginScreen extends Component {
  constructor (props) {
    super(props)

    this.state = {
      email: '',
      password: '',
      response: ''
    }

    this.po = new PostOffice()
  }

  async signup () {
    DismissKeyboard()

    try {
      await firebase.auth().createUserWithEmailAndPassword(this.state.email, this.state.password)

      this.setState({
        response: 'account created'
      })

      this.po.getPacket('user').email = this.state.email
    } catch (error) {
      this.setState({
        response: error.toString()
      })
    }
  }

  async login () {
    DismissKeyboard()

    try {
      await firebase.auth().signInWithEmailAndPassword(this.state.email, this.state.password)

      this.setState({
        response: 'Logged In!'
      })

      this.po.getPacket('user').email = this.state.email
    } catch (error) {
      this.setState({
        response: error.toString()
      })
    }
  }

  render () {
    return (
      <View style={styles.container}>
        <Header title='Login/Signup' menuButtonPress={this.props.menuButtonPress} />
        <View>
          <Input
            label='Email'
            value={this.state.email}
            onChangeText={email => this.setState({ email })}
            placeholder='user@gmail.com'
          />
        </View>
        <View>
          <Input
            label='Password'
            value={this.state.password}
            onChangeText={password => this.setState({ password })}
            placeholder='your password'
            secureTextEntry
        />
        </View>
        <View style={styles.buttons}>
          <Button onPress={() => this.login()}>
          Log in
          </Button>
          <Button onPress={() => this.signup()}>
          Sign up
          </Button>
        </View>
        <View style={styles.response}>
          <Text style={styles.responseText}>{this.state.response}</Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Theme.backgroundColor
  },

  buttons: {
    marginTop: 50
  },

  responseText: {
    textAlign: 'center',
    width: '100%',
    fontSize: 18,
    color: 'white'
  },

  response: {
    marginTop: 50
  }

})

const mapStateToProps = (state) => {
  return {}
}

const mapDispatchToProps = (dispatch) => {
  return {
    menuButtonPress: () => { dispatch(Actions.drawerOpen()) }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen)
