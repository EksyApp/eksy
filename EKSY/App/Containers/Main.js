import React, {Component} from 'react'
import NavigationRouter from '../Navigation/NavigationRouter'
import FirebaseConfig from '../includes/FirebaseConfig'
import {Provider} from 'react-redux'
import {ActivityIndicator} from 'react-native'
import Store from '../Store'

class Main extends Component {

  state = {
    store: null
  }

  constructor (props) {
    super(props)
    FirebaseConfig.initialise()
  }

  async componentWillMount() {
    const store = await Store()
    this.setState({ store })
  }

  render () {
    if (!this.state.store) {
      return (
        <ActivityIndicator />
      )
    }

    return (
      <Provider store={this.state.store}>
        <NavigationRouter />
      </Provider>
    )
  }
}

export default Main
