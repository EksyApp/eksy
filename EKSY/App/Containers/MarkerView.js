import React, {Component} from 'react'
import {View, ScrollView, Text, StyleSheet} from 'react-native'
import Header from "../Components/Header";
import PictureList from '../Components/PictureList'
import * as ReactActions from '../Actions'
import {connect} from 'react-redux'
import TextArea from '../Components/TextArea'
import * as Theme from '../Theme'
import Divider from '../Components/Divider'
import Label from '../Components/Label'

export class MarkerView extends Component {

	_renderImages() {
		if(this.props.marker.images.length > 0) {
			return(
					<View>
						<Divider />
						<Label>Images</Label>
						<PictureList data={this.props.marker.images} listStyle={style.listStyle} imageContainerStyle={style.imageContainer} />
					</View>
			)
		} else {
			return null
		}
	}

	render() {
		return(
				<View style={style.container}>
					<Header backButton title={this.props.marker.title}/>
					<ScrollView>
						<View>
							<TextArea>
								{this.props.marker.text}
							</TextArea>
						</View>

						{this._renderImages()}

					</ScrollView>
				</View>
		)
	}

}

const style = StyleSheet.create({
	imageContainer: {
		width: '100%',
	},

	listStyle: {
		margin: 10
	},

	container: {
		backgroundColor: Theme.backgroundColor,
		flex: 1
	},

})

const mapStateToProps = (state) => {
	return {
		marker: state.map.selectedMarker.marker
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MarkerView)
