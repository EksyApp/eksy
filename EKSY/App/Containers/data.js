export const characters = [
  {
    name: 'Kevin',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane03/sub-buzz-3484-1481210659-5.png',
    coordinate: [ 40.764466, -73.974488 ],
    good: true,
  },
  {
    name: 'Uncle Frank',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane02/sub-buzz-25788-1481210705-7.png',
    coordinate: [ 40.764656, -73.980907 ],
    good: false,
  },
  {
    name: 'Aunt Leslie',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane01/sub-buzz-5211-1481210889-4.png',
    coordinate: [ 40.764838, -73.981193 ],
    good: false,
  },
  {
    name: 'The guy with the same coat as Kevin’s dad',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane03/sub-buzz-4392-1481212051-3.jpg',
    coordinate: [ 40.748418, -73.985702 ],
    good: false,
  },
  {
    name: 'Donald Trump',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane03/sub-buzz-5109-1481213313-1.png',
    coordinate: [ 40.762389, -73.973817 ],
    good: false,
  },
  {
    name: 'The lady from Plaza Hotel reservations',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-6769-1481213418-3.png',
    coordinate: [ 40.779707, -73.979012 ],
    good: false,
  },
  {
    name: 'This scary ass big-tilted Santa',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane02/sub-buzz-27082-1481212691-2.png',
    coordinate: [ 40.747067, -74.004496 ],
    good: false,
  },
  {
    name: 'Random cousins/siblings',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-7903-1481215701-1.png',
    coordinate: [ 40.765298, -73.976022 ],
    good: false,
  },
  {
    name: '“Watch it kid HAHAHAHA” guy',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-8357-1481215914-1.png',
    coordinate: [ 40.791939, -73.963466 ],
    good: false,
  },
  {
    name: 'The taxi driver near Central Park',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-8061-1481216019-8.png',
    coordinate: [ 40.792364, -73.964200 ],
    good: false,
  },
  {
    name: 'The prostitutes near Central Park',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane03/sub-buzz-6355-1481215967-1.png',
    coordinate: [ 40.791646, -73.964681 ],
    good: false,
  },
  {
    name: 'Buzz',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane03/sub-buzz-4009-1481211103-1.png',
    coordinate: [ 40.764298, -73.971752 ],
    good: false,
  },
  {
    name: 'Esther Stone, hotel lady',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-6874-1481213534-3.png',
    coordinate: [ 40.764116, -73.973581 ],
    good: false,
  },
  {
    name: 'Fuller',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane02/sub-buzz-26273-1481211159-3.png',
    coordinate: [ 40.763241, -73.975782 ],
    good: false,
  },
  {
    name: 'Mrs. McCallister',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane02/sub-buzz-25796-1481211030-8.png',
    coordinate: [ 40.758702, -73.978598 ],
    good: false,
  },
  {
    name: 'Mr. McCallister',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane03/sub-buzz-4020-1481211789-2.png',
    coordinate: [ 40.758889, -73.985125 ],
    good: false,
  },
  {
    name: 'The NYC cop who takes Mrs. McCallister to Rockefeller Plaza',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/13/asset/buzzfeed-prod-fastlane01/sub-buzz-21202-1481220551-1.png',
    coordinate: [ 40.771156, -73.979802 ],
    good: true,
  },
  {
    name: 'The woman that slaps/punches Marv twice',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane03/sub-buzz-5073-1481213214-2.png',
    coordinate: [ 40.761432, -73.977645 ],
    good: false,
  },
  {
    name: 'Harry',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-6492-1481212963-3.png',
    coordinate: [ 40.792084, -73.966925 ],
    good: false,
  },
  {
    name: 'Tim Curry',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-6839-1481213501-1.png',
    coordinate: [ 40.764859, -73.975421 ],
    good: true,
  },
  {
    name: 'Duncan',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane02/sub-buzz-28268-1481214964-2.png',
    coordinate: [ 40.763560, -73.972198 ],
    good: true,
  },
  {
    name: 'Marv',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane02/sub-buzz-27059-1481212962-1.png',
    coordinate: [ 40.791871, -73.966812 ],
    good: false,
  },
  {
    name: 'The Pigeon Lady',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane02/sub-buzz-27109-1481213135-7.png',
    coordinate: [ 40.765126, -73.979924 ],
    good: true,
  },
];
