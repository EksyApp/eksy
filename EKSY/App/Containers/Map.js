import React, { Component } from 'react';
import {
  StyleSheet,         // CSS-like styles
  Text,               // Renders text
  TouchableOpacity,   // Pressable container
  View                // Container component
} from 'react-native';
import MapView from 'react-native-maps';
// Import data
import { characters } from './data';
import Callout from './Callout';
import styles from './Styles/MapStyles'

export default class Map extends Component {

  state = {
    // Show good or all characters flag
    showGoodOnly: false,
  }

  render() {
    return (
      <View style={styles.container}>
      {/* Map*/}
      <MapView
      style={styles.map}
      // Position on Manhattan, New York
      initialRegion={{
        latitude: 40.77096,
        longitude: -73.97702,
        latitudeDelta: 0.0491,
        longitudeDelta: 0.0375,
      }}
      >
      {/* Loop through characters and add pins on the map */}
      {characters.map((character, index) =>
            // If showGoodOnly is true, but the character is bad - do not show it
            this.state.showGoodOnly && !character.good || <MapView.Marker
              coordinate={{
                latitude: character.coordinate[0],
                longitude: character.coordinate[1],
              }}
              // Callout offset
              calloutOffset={{ x: -8, y: 28 }}
              // Greed color for good characters and red for others
              pinColor={character.good ? '#009688' : '#f44336'}
              key={index}
            >
              {/* Callout */}
              <MapView.Callout tooltip style={styles.callout}>
                <Callout
                  name={character.name}
                  image={character.image}
                />
              </MapView.Callout>
            </MapView.Marker>
          )}
      </MapView>
      {/* Button */}
      <View style={styles.buttonContainer}>
      <TouchableOpacity
      style={styles.button}
      // Toggle this.state.showGoodOnly
      onPress={() => this.setState({
        showGoodOnly: !this.state.showGoodOnly
      })}
      >
      <Text>{this.state.showGoodOnly ? 'Show All' : 'Show Good Only'}</Text>
      </TouchableOpacity>
      </View>
      </View>
    );
  }
}
