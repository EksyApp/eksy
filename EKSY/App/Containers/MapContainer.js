import React, {Component} from 'react'
import {PermissionsAndroid} from 'react-native'
import Map from './Map/Map'
import MenuButton from '../Components/MenuButton'
import {View, StyleSheet, Dimensions, Animated, Text, TouchableWithoutFeedback} from 'react-native'
import {Badge} from 'react-native-elements'
import * as Actions from '../Actions'
import {connect} from 'react-redux'
import Interactable from 'react-native-interactable'
import MarkerCarousel from '../Components/MarkerCarousel'
import {backgroundColor, detailColor} from '../Theme'
import MarkerView from '../Containers/MarkerView'

const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}

export class MapContainer extends Component {

	constructor(props) {
		super(props)

		this._deltaY = new Animated.Value(Screen.height - 100);
	}

	async requestLocationPermission() {
		try {
			const granted = await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION)
			if (granted !== PermissionsAndroid.RESULTS.GRANTED) {
				console.log('no permissions granted') // prompt user?
			}
		} catch (err) {
			console.warn(err)
		}
	}

	async componentWillMount() {
		await this.requestLocationPermission()
	}

	render() {
		return (
				<View style={styles.viewContainer}>
					<Map
							currentRegion={this.props.currentRegion}
							currentLocation={this.props.currentLocation}
							markerList={this.props.markerList}
							setMarkerSelected={this.props.setMarkerSelected}
							setMarkerViewVisible={this.props.setMarkerViewVisible}
							disableGestures={this.props.disableGestures}
							regionChange={this.props.regionChange}/>
					<MenuButton onPress={() => {
						this.props.menuButtonPress()
					}}/>
					<View style={styles.panelContainer}>
						<Animated.View style={[styles.panelContainer, {
							backgroundColor: 'black',
							opacity: this._deltaY.interpolate({
								inputRange: [0, Screen.height - 100],
								outputRange: [1, 0],
								extrapolateRight: 'clamp'
							})
						}]}
						pointerEvents="none"/>
						<Interactable.View
								verticalOnly={true}
								snapPoints={[{y: Screen.height - 220}, {y: Screen.height}]}
								boundaries={{top: 300}}
								initialPosition={{y: Screen.height}}
								animatedValueY={this._deltaY}>
							<Animated.View style={styles.panel}>
								<Animated.View style={styles.panelHeader}>
									<Animated.View style={styles.panelHandle}/>
									{this.props.markerList.length > 0 &&
									<Badge
											value={this.props.markerList.length}
											containerStyle={styles.badgeContainer}
											textStyle={styles.badgeText}
									/>
									}
								</Animated.View>
								<View style={styles.panelVisible}>
									<MarkerCarousel
											markerList={this.props.markerList}
											setMarkerSelected={this.props.setMarkerSelected}
											setMarkerViewVisible={this.props.setMarkerViewVisible}
											disableGestures={this.props.disableGestures}
											pointerEvents="none"/>
								</View>

							</Animated.View>
						</Interactable.View>
					</View>

					<MarkerView />
				</View>
		)
	}
}

// <View style={styles.panel}>
// </View>
const styles = StyleSheet.create({
	container: {
		flex: 1
	},
	panelContainer: {
		position: 'absolute',
		top: 0,
		bottom: 0,
		left: 0,
		right: 0,
		flex: 1,
	},
	viewContainer: {
		flex: 1,
	},
	panel: {
		width: Screen.width,
		height: Screen.height,
	},

	panelVisible: {
		paddingTop: 20,
		width:"100%",
		height: Screen.height,
		backgroundColor: backgroundColor,
		borderTopLeftRadius: 20,
		borderTopRightRadius: 20,
		shadowColor: '#000000',
		shadowOffset: {width: 0, height: 0},
		shadowRadius: 5,
		shadowOpacity: 0.4
	},

	panelHeader: {
		alignItems: 'center',
		position: 'relative',
		height: 20
	},
	panelHandle: {
		width: 40,
		height: 8,
		borderRadius: 4,
		backgroundColor: detailColor,
		top: 30,
		zIndex: 1,
		position: "absolute"
	},
	badgeContainer: {
		backgroundColor: '#800000',
		position: 'relative',
		top: 22,
		left: 50
	},
	badgeText: {
		color: '#FFFFFF'
	}
})

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion,
		currentLocation: state.map.location,
		markerList: state.markers.markerList,
		markerViewVisible: state.ui.markerView.markerViewVisible
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		menuButtonPress: () => {
			dispatch(Actions.drawerOpen())
		},
		regionChange: (region) => {
			dispatch(Actions.updateRegion(region))
		},
		setMarkerSelected: (marker) => {
			dispatch(Actions.setMarkerSelected(marker))
		},
		setMarkerViewVisible: () => {
			dispatch(Actions.setMarkerViewVisible())
		},
		setMarkerViewHidden: () => {
			dispatch(Actions.setMarkerViewHidden())
		},
		disableGestures: (value) => {
			dispatch(Actions.disableGestures(value))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MapContainer)
