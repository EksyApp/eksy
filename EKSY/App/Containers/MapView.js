import React, {Component} from 'react'
import { Container, Content, Button, Icon } from 'native-base'
import {PermissionsAndroid} from 'react-native'
import Map from '../Map/Map'
import Styles from './Styles/MapViewStyles'
import MenuButton from '../Components/MenuButton'
import MapManager from '../Map/MapManager'
import Marker from '../Map/Marker'
import {View} from 'react-native'

class MapView extends Component {

  constructor (props) {
    super(props)

    this.requestLocationPermission()

    // let marker = new Marker(60.185359, 24.951338, 1);
    // marker.setText("Testi text");
    // new MapManager().addMarker(marker);
  }

  async requestLocationPermission () {
    try {
      await PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION)
    } catch (err) {
      console.warn(err)
    }
  }

  render () {
    return (
      <View style={Styles.container}>
        <Map />
        <MenuButton transparent />
      </View>
    )
  }
}

export default MapView
