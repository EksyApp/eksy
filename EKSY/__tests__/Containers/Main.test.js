import React from 'react'
import { Main } from '../../App/Containers/Main'
import renderer from 'react-test-renderer'

jest.mock('react-native-version-number', () => {
	return {
		appversion: 'test'
	}
})

jest.mock('react-native-code-push', () => {
	function MockCodePush(options = {}) {
		return jest.fn()
	}
	
	Object.assign(MockCodePush, {
		sync: jest.fn(),
		installMode: null,
		updateDialog: null,
		CheckFrequency: {
			ON_APP_RESUME: null
		},
		InstallMode: {
			IMMEDIATE: null
		}
	})
	
	return MockCodePush
})

jest.mock('../../App/Containers/Settings/FilterSettingsContainer', () => {
	return require('react-native').View
})

jest.mock('../../App/Containers/Settings/ProfileContainer', () => {
	return require('react-native').View
})

jest.mock('react-native-splash-screen', () => {
	return {
		hide: () => {}
	}
})

test('Main renders correctly', () => {
	const tree = renderer.create(
			<Main
			/>
	).toJSON();
	expect(tree).toMatchSnapshot();
});