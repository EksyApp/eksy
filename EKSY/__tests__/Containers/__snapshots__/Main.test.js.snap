// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="large"
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
/>
`;
